micropip/package_index.py

```diff
@@ -182,3 +190,59 @@ def _fast_check_incompatibility(filename: str) -> bool:
         return False
 
     return True
+
+
+def _check_index_url(url: str) -> None:
+    fields = [parsed[1] for parsed in _formatter.parse(url)]
+
+    if "package_name" not in fields:
+        raise ValueError(
+            f"Invalid index URL: {url!r}. "
+            "Please make sure it contains the placeholder {package_name}."
+        )
+
+
+async def search_packages(
+    pkg: str,
```

Comment:
Should it be `search_package` if it searches just 1 package? Also, it might be more user friendly to name the first argument `name`. 

It's not "search" technically though. Search is returning top k results for a query, possibly with some pre-processing or fuzzing of the query. While this return a single response with which is the exact match. Maybe `query_package`? But I'm open to other name suggestions.

Reply:
Thanks for the suggestion! I wasn't aware of the difference between search and query :) I will go with `query_package`.